File: C2/PDOO/Practicas/DeepSpaceRuby/lib/NumericDamageToUI.rb

require_relative 'DamageToUI'

module Deepspace

# 2.3.17 - Translation from Java
# @author Profesor
    
class NumericDamageToUI < DamageToUI
  
  public_class_method :new
  
  attr_reader :nWeapons
  
  def initialize (d)
    super(d)
    @nWeapons=d.nWeapons
  end
    
  public
  
  #Override
  def getWeaponInfo() 
    return @nWeapons.to_s
  end
  
end # class

end # module

File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/NumericDamageToUI.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Profesor
 */
public class NumericDamageToUI extends DamageToUI{
    private int nWeapons;

    NumericDamageToUI(NumericDamage d) {
        super(d);
        nWeapons=d.getNWeapons();
    }

    public int getNWeapons() {
        return nWeapons;
    } 
    
    @Override
    public String getWeaponInfo() {
        return ""+nWeapons;
    }
}
